src/GridFS/CollectionWrapper.php

```diff
@@ -80,6 +80,30 @@ public function deleteChunksByFilesId($id): void
         $this->chunksCollection->deleteMany(['files_id' => $id]);
     }
 
+    /**
+     * Delete all GridFS files and chunks for a given filename.
+     */
+    public function deleteFileAndChunksByFilename(string $filename): int
+    {
+        /** @var iterable<array{_id: mixed}> $files */
+        $files = $this->findFiles(['filename' => $filename], [
+            'codec' => null,
+            'typeMap' => ['root' => 'array'],
+            'projection' => ['_id' => 1],
+        ]);
+
+        /** @var list<mixed> $ids */
```

Comment:
It's a shame [`array_column()`](https://www.php.net/array_column) doesn't support iterables (despite allowing elements _within_ the input array to be either arrays or objects).

Reply:
Yes, I tried. We need the function `iterable_column`